inc/saitam.h

```diff
@@ -1472,9 +1514,44 @@ typedef enum _sai_tam_telemetry_attr_t
      * @type sai_uint32_t
      * @flags CREATE_AND_SET
      * @default 1
+     * @validonly SAI_TAM_TELEMETRY_ATTR_TAM_REPORTING_TYPE == SAI_TAM_REPORTING_TYPE_TIME_BASED
      */
     SAI_TAM_TELEMETRY_ATTR_REPORTING_INTERVAL,
 
+    /**
+     * @brief Tam telemetry reporting type
+     *
+     * @type sai_tam_reporting_type_t
+     * @flags CREATE_AND_SET
+     * @default SAI_TAM_REPORTING_TYPE_TIME_BASED
+     */
+    SAI_TAM_TELEMETRY_ATTR_TAM_REPORTING_TYPE,
+
+    /**
+     * @brief Tam telemetry reporting chunk size
+     *
+     * defines the size of reporting chunk, which means TAM will report to the collector every time
+     * if the report count reaches the chunk size.
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 1
+     * @validonly SAI_TAM_TELEMETRY_ATTR_TAM_REPORTING_TYPE == SAI_TAM_REPORTING_TYPE_COUNT_BASED
+     */
+    SAI_TAM_TELEMETRY_ATTR_REPORTING_CHUNK_SIZE,
+
+    /**
+     * @brief Tam telemetry cache size
+     *
+     * If the collector isn't ready to receive the report, this value indicates how many
+     * reports that can be cached. 0 means no cache which is the default behavior.
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_TELEMETRY_ATTR_CACHE_SIZE,
```

Comment:
since the ring buffer is created on driver load, we need to move it to sai switch attribute.